### ISSUE TITLE
Weird letter spacing for 0.43.0 vs 0.42.0 | Answer: Use monospace font families

### ISSUE THREAD

**Issue by do-kevin** [sedwards2009/extraterm#206]
Is it just my installation or is anyone else seeing this? My font family and font size are the same between both versions.

0.43.0
![0 43 0](https://user-images.githubusercontent.com/40156784/63231228-909a5280-c1cd-11e9-965b-80338366a5be.png)

0.42.2
![0 42 2](https://user-images.githubusercontent.com/40156784/63231232-95f79d00-c1cd-11e9-8bec-46c829fdc219.png)

**Comment by cobrabr** [comment]
Same issue here. Every time I reopen Extraterm, the font looks like the first screenshot. Also, on mine, there's a weird, thick border around the frame, which didn't use to be there before. Using it on Win10 x64. I'll grab a screenshot later at home (at work, I'm still on 0.42, exactly for those 2 reasons).

**Comment by sedwards2009** [comment]
@do-kevin  The problem is that in a terminal you are meant to use a monospace font because all of the rows and columns of characters are supposed to line up neatly. You're using a proportional font where the width of the character varies. The bug is in the 0.42.2 screenshot. Extraterm used to assume that all characters had the same width. Since the change in how the text is drawn, that assumption is fixed.

So basically, no bug, but you should choose a monospace font, like Consolas or Dejavu sans mono.

**Comment by sedwards2009** [comment]
@cobrabr  do post a screenshot here so that we check whether it is the same issue.

**Comment by do-kevin** [comment]
> 
> 
> Same issue here. Every time I reopen Extraterm, the font looks like the first screenshot. Also, on mine, there's a weird, thick border around the frame, which didn't use to be there before. Using it on Win10 x64. I'll grab a screenshot later at home (at work, I'm still on 0.42, exactly for those 2 reasons).

Could you provide a screenshot so we could see that thick border? I think it could actually be the margins.

> 
> 
> @do-kevin The problem is that in a terminal you are meant to use a monospace font because all of the rows and columns of characters are supposed to line up neatly. You're using a proportional font where the width of the character varies. The bug is in the 0.42.2 screenshot. Extraterm used to assume that all characters had the same width. Since the change in how the text is drawn, that assumption is fixed.
> 
> So basically, no bug, but you should choose a monospace font, like Consolas or Dejavu sans mono.

Okay, I'll test it out. Thanks seward.

**Comment by cobrabr** [comment]
I'll post the screenshot later, but in my case, I was already using a monospace font (tested with Oxygen Mono and Fira Code), and I had the same issue.

**Comment by cobrabr** [comment]
Here is the 0.42.2 screenshot (which is working fine):

![image](https://user-images.githubusercontent.com/724969/63298990-c4937780-c2a2-11e9-9b02-ac8c607fb5b0.png)

**Comment by cobrabr** [comment]
Here is the screenshot for 0.43: 
![image](https://user-images.githubusercontent.com/724969/63313191-a5f8a500-c2d1-11e9-8ec2-d8a4fe63a602.png)

Notice the font and the grey border around the frame. The configuration is the same as the one from the screenshot above.

**Comment by sedwards2009** [comment]
@cobrabr Thanks for that. I've seen this before on macOS but no where else until now. The background colour of the terminal itself seems to change for no reason. I'll try poking at it on Windows to see if I can find a way of reproducing this problem on demand. Then I have a chance of figuring it out.

**Comment by sedwards2009** [comment]
@cobrabr  Looking at your screenshots, it appears that there is some tweaking to do with respect to the character spacing. It looks a bit too wide and big on the 0.43.

**Comment by cobrabr** [comment]
Thanks for the quick reply, @sedwards2009, I appreciate it. :)

When I change the font back to Oxygen Mono (which only lasts until I close the application; it reverts back to the font on my screenshot every time I open it), the spacing looks _about_ the same as on 0.42.2 (i.e. it wasn't noticeable/didn't bother me). With the "weird" font, yeah, the spacing is all over the place, but I'm thinking it's the font itself more than Extraterm.

**Comment by cobrabr** [comment]
And now that you mention it (and looking at both screenshots), you're right, it's not the border that's the problem, it's the terminal background colour that changes incorrectly.

**Comment by cobrabr** [comment]
@sedwards2009, FYI 0.44 still has the same problem. Should I open a new issue for it?

**Comment by sedwards2009** [comment]
@cobrabr  This issue is fine. I still have to look into that Oxygen Mono problem.

**Comment by sedwards2009** [comment]
@cobrabr  I was just able to reproduce the totally-wrong-font problem here.

Oxygen Mono looks good when it works.

**Comment by cobrabr** [comment]
Thanks again, @sedwards2009. Did you manage to find the reason behind the background colour changing as well?

**Comment by sedwards2009** [comment]
Fixed in v0.44.1.

**Comment by sedwards2009** [comment]
@cobrabr  That background color thing is on my TODO list. I've got a bunch of other little things which need attention.

**Comment by cobrabr** [comment]
@sedwards2009 Thanks for the heads-up. Do you need me to open a new issue for it?

Also, thanks for fixing this one! 😄

**Comment by sedwards2009** [comment]
@cobrabr  please do
